renderer/src/hooks/StationWallet.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect } from 'react'
+import {
+  getDestinationWalletAddress,
+  setDestinationWalletAddress,
+  getStationWalletAddress,
+  getStationWalletBalance,
+  getStationWalletTransactionsHistory
+} from '../lib/station-config'
+import { FILTransaction } from '../typings'
+
+const useWallet = (): [string, string | undefined, number, FILTransaction[] | [], (address: string|undefined) => void, FILTransaction | undefined, () => void] => {
+  const datt = Date.now()
+  const [stationAddress, setStationAddress] = useState<string>('')
+  const [destinationFilAddress, setDestinationFilAddress] = useState<string | undefined>()
+  const [walletBalance, setWalletBalance] = useState<number>(0)
+  const [walletTransactions, setWalletTransactions] = useState<FILTransaction[]>([])
+  const [currentTransaction, setCurrentTransaction] = useState<FILTransaction | undefined>()
+
+  const setDestinationAddress = async (address: string | undefined) => {
+    await setDestinationWalletAddress(address)
+    setDestinationFilAddress(address)
+  }
+
+  const dismissCurrentTransaction = () => {
+    if (currentTransaction && currentTransaction.status !== 'processing') {
+      setCurrentTransaction(undefined)
+      setWalletTransactions([currentTransaction, ...walletTransactions])
+    }
+  }
+
+  useEffect(() => {
+    const loadStoredInfo = async () => {
+      setDestinationFilAddress(await getDestinationWalletAddress())
+      setStationAddress(await getStationWalletAddress())
+      setWalletBalance(await getStationWalletBalance())
+      setWalletTransactions(await getStationWalletTransactionsHistory())
+    }
```

Comment:
Maybe we could make this run in parallel if they are not required to wait on each other.